integration-cli/docker_cli_build_test.go

```diff
@@ -602,6 +602,34 @@ RUN [ $(cat "/test dir/test_file6") = 'test6' ]`, command, command, command, com
 	}
 }
 
+func (s *DockerSuite) TestBuildChownFlagAdd(c *check.C) {
+	testRequires(c, DaemonIsLinux) // Linux specific test
+	buildImageSuccessfully(c, "testaddwithchown", build.WithBuildContext(c,
+		build.WithFile("Dockerfile", fmt.Sprintf(`FROM busybox
+RUN echo 'test1:x:1001:1001::/bin:/bin/false' >> /etc/passwd
+RUN echo 'test1:x:1001:' >> /etc/group
+RUN echo 'test2:x:1002:' >> /etc/group
+ADD --chown=test1:1002 . /new_dir
+RUN ls -l /
+RUN [ $(ls -l / | grep new_dir | awk '{print $3":"$4}') = 'test1:test2' ]`)),
```

Comment:
I would be inclined to add a test that does `ls -ln` to chec the numberi values too as well as the names...